python-checks/src/test/resources/checks/numpyListOverGenerator.py

```diff
@@ -0,0 +1,35 @@
+from typing import Any
+
+
+def import_1():
+    import numpy as np
+    from typing import Any
+
+    gen_1 = (x * 2 for x in range(5))
+    arr = np.array(gen_1)  # Noncompliant {{Pass a list to "np.array" instead of passing a generator.}}
+    #     ^^^^^^^^^^^^^^^
+
+    gen_2: Any = (x * 2 for x in range(5))  # This is a false negative. ReachingDefinitionsAnalysis does not find the variable when it has
+    # been type annotated. Let's construct a ticket for this.
+    arr = np.array(gen_2)
```

Comment:
I think we should change a bit the comment here. Because we do not want to raise an issue when the generator is type-hinted to `Any`. We should maybe add another line with another `gen_2` being assigned without a type hint. for example:
```
  gen_2 = (x * 2 for x in range(5))  # Here would be the real FN
  arr = np.array(gen_2)
``` 
Because of the constraints of ReachingDefinitionAnalysis we do not raise an issue here.
Adding this test will help when fixing the issue because it will fail if ReachingDefinitionAnalysis can handle the type hints. 

I think we can also remove the comment about the ticket, we can just create it. Here is the ticket:(https://sonarsource.atlassian.net/browse/SONARPY-1470).

Reply:
Addressed in 05740032e5e18f3cbf6e22f821f61d8f5c9f7f1f.